packages/apps/shopify-app-remix/src/server/authenticate/admin/scope/__tests__/query.test.ts

```diff
@@ -25,9 +25,13 @@ it('returns scopes information', async () => {
 
   // THEN
   expect(result).not.toBeUndefined();
-  expect(result.granted).toEqual(['read_orders', 'write_customers']);
+  expect(result.granted).toEqual([
+    'read_orders',
+    'write_customers',
+    'read_customers',
+  ]);
   expect(result.required).toEqual(['read_orders', 'read_reports']);
-  expect(result.optional).toEqual(['write_customers']);
+  expect(result.optional).toEqual(['write_customers', 'read_customers']);
```

Comment:
It may be helpful to revisit our [mocked responses](https://github.com/Shopify/shopify-app-js/blob/7635b71da2c18ea2fc15c7b70ed94512207e9a2b/packages/apps/shopify-app-remix/src/server/authenticate/admin/scope/__tests__/mock-responses.ts) as well.

Reply:
While updating coverage, I realized there might be an edge case. Do we wish to return required scopes as implied optional ones as well? 

For example, if we request explicitly for "read_customers" and set "write customers" as an optional scope, and then both are granted. Under the changes I have made, we would return "read_customers" in `required` but also as an implicit scope in `optional`.

Reply:
I have corrected the Remix query behaviour to return exactly what Admin API returns. 
 
 I have also corrected the original test coverage and mock calls and added new scopes to demonstrate functionality.
 
 - `read_reports` has been added to the `installedAccessScopes` as it is a required scope
 -  `read_customers` has been added to the `installedAccessScopes` as it is a implied scope of the installed `write_customers`
 -   `write_products` has been added to `optionalAccessScopes` and `read_products` has been added to `requiredAccessScopes` and `installedAccessScopes`, to ensure that, like with the Admin API, the Remix API does not return required scopes as implied in optional scopes.
 
![image](https://github.com/user-attachments/assets/b60f3bf2-761b-4c48-a3a4-1b6fb1fd9f06)

 
![image](https://github.com/user-attachments/assets/76abd01a-0261-4317-838b-17fe0a11f1e2)